libbeat/processors/fingerprint/hash.go

```diff
@@ -34,6 +36,7 @@ var hashes = map[string]hashMethod{
 	"sha256": sha256.New,
 	"sha384": sha512.New384,
 	"sha512": sha512.New,
+	"xxhash": func() hash.Hash { return xxhash.New64() },
```

Comment:
I think you can just do this?

```suggestion
	"xxhash": xxhash.New64,
```

Reply:
No sure about that, does that compile?

Reply:
Hmm, you're right, it doesn't compile because `xxhash.New64` doesn't return a `hash.Hash`. Let me look into the `xxhash` package implementation a bit more as to why this is the case.

Reply:
So it looks like the [`*XXHash64` type](https://github.com/OneOfOne/xxhash/blob/8f0be54a8d5ccf68817143d8c996147ec5cbd795/xxhash.go#L94) being returned by [`xxhash.New64`](https://github.com/OneOfOne/xxhash/blob/8f0be54a8d5ccf68817143d8c996147ec5cbd795/xxhash.go#L125) doesn't implement the [`hash.Hash`](https://golang.org/pkg/hash/#Hash) interface completely 😞. Specifically it seems to be missing the `Write` method, which is specified by the `io.Writer` interface embedded within the `hash.Hash` interface. I wonder why that is 🤔. This is especially concerning to me because the fingerprint processor code does [use](https://github.com/elastic/beats/blob/77019e7a1f28929a1bcc44362c68fdbcb216a8dc/libbeat/processors/fingerprint/fingerprint.go#L87) the `io.Writer` interface.

Would you mind filing an issue in https://github.com/OneOfOne/xxhash about this? If not, I can do it. I'd like to see what they have to say before we make any decisions on how to proceed on this PR here.

Reply:
The type `*XXHash64` implements completely the interface `hash.Hash`. What it is not correct is the signature of the function `func New64() *XXHash64` that it is different from `type hashMethod func() hash.Hash`, so this is the reason of the wrapper function or inline function to provide the xxhash method.

I think it is not a problem, we are providing a `hash.Hash` implementation with `*XXHash64`.

Reply:
For now lets put a comment here as to why we need the wrapper function and can't use `xxhash.New64` directly. I created an issue in the xxhash repo asking if they can make the construction functions return `hash.Hash`: https://github.com/OneOfOne/xxhash/issues/29. Please refer to this in the comment so we know to come fix the implementation here if they make changes on their end. Thanks!

Reply:
Given that this comment might get long, it might make sense now to move the function back to how you had it before (not inline).

Reply:
OneOfOne/xxhash#29 has been resolved already and new version of the package, v1.2.7, has been released. It exports two new constructor functions, `NewHash32` and `NewHash64` which return `hash.Hash`. 

So we should be able to vendor the new release and just use `xxhash.NewHash64` inline here without needing the wrapper any more!

Reply:
Ok I have updated xxhash dependency to `v1.2.7` :+1: